opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/RelevanceQuery.java

```diff
@@ -31,26 +27,43 @@
 
   @Override
   public QueryBuilder build(FunctionExpression func) {
-    List<Expression> arguments = func.getArguments();
+    var arguments = func.getArguments().stream()
+        .map(a -> (NamedArgumentExpression)a).collect(Collectors.toList());
     if (arguments.size() < 2) {
       throw new SyntaxCheckException(
           String.format("%s requires at least two parameters", getQueryName()));
     }
-    NamedArgumentExpression field = (NamedArgumentExpression) arguments.get(0);
-    NamedArgumentExpression query = (NamedArgumentExpression) arguments.get(1);
+
+    // Aggregate parameters by name, so getting a Map<Name:String, List>
+    arguments.stream().collect(Collectors.groupingBy(a -> a.getArgName().toLowerCase()))
+        .forEach((k, v) -> {
+          if (v.size() > 1) {
+            throw new SemanticCheckException(
+                String.format("Parameter '%s' can only be specified once.", k));
+          }
+        });
```

Comment:
Do we need to aggregate all duplicates into one exception? Currently, function throws on first found duplicated parameter.

Reply:
What do we do other places...?  I think we always throw on the first failure, but maybe you have an example where we aggregate?

Reply:
No examples. I don't think that we should worry about that, the probability that a user will specify multiple arguments more than once is extremely low I suppose. I just pointed that we have a room for further improvement.